WebInstall/explainer.md

```diff
@@ -112,61 +156,79 @@ Additionally for developers, the feature also:
+
+* Resolve when an installation was completed.
+    * The success value will be an object that contains:
+     	*  `mode`: string with the surface-hint where the app was installed.
+* Be rejected if the prompt is not shown or if the app installation did not complete. It'll reject with a [`DOMException`](https://developer.mozilla.org/en-US/docs/Web/API/DOMException) value of:
+    * `NotAllowedError`: The `installation` [Permissions Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Permissions_Policy) has been used to block the use of this feature.
+    * `NotSupportedError`: the target website is not installable.
+    * `InsufficientEngagementError`: the UA's required (if any) [engagement heuristics](https://web.dev/install-criteria/#criteria) have not been met.
+    * `AbortError`: The installation (prompt) was closed/cancelled.
+    * `TimeoutError`: The installation failed due to timeout.
+    * `OperationError`: other error.
+	
+    
+ ```javascript
+/* simple example of using navigator.install */
+
+const installApp = async () => {
+    try{
+        const value = await navigator.install();
+    }
+    catch(err){console.error(err.message)}
+};
 
-if ('install' in navigator ) {
-    navigator.install('https://www.elk.zone')
-    .then( origin => console.log( `Installation started from ${origin}` ));
-}
 ```
 
-* Be rejected if the prompt is not shown and have a `reason` value of:
-    * "disallowed": `String` indicating the origin website trying to install the web app does not have install permissions.
-    * "not-installable": `String` indicating that the target website is not installable.
-    * *error*: instance of `Error` providing details about an error that happened.
- 
 ```javascript
+/* Example of advanced error handling */
+
 (...)
-.catch(reason => {
-    if(reason === "not-installable") {
+.catch(error => {
+    if(error.name === 'NotSupportedError') {
         // target site is not installable
         console.log("Website is not an installable app.");
     }
-    else if (reason === "disallowed") {
+    else if (error.name === 'NotAllowedError') {
         // origin cannot install other webapps.
         console.log("Origin does not have permissions to install web apps.");
     }
     else {
-        console.log(reason);
+        console.log(error.message);
     }
 });
 ```
 
-Resolution of the promise does not indicate that the app was installed. It only indicates that the installation prompt has been shown. This is a design decision to preserve privacy.
+![Promises resolve/reject flow](./installPromises.png) 
+
+Upon a successful installation there is **one** opportunity for the installation origin to get attribution for the install, parse any additional information like ad campaigns, or any other required processing in the origin after a successful installation. Once the promise resolves there is no other way for the origin to get access to the same information.* 
```

Comment:
Details about the campaign id were removed, so does this bit about attribution still apply? Or should the campaign id be added back?

Reply:
campaign id will now go as the referral info parameter . it is no longer returned to the origin, as the origin has no use for this info returned!

Reply:
This is still worded as if campaign id/referral info is handled _after_ installation. Is the intention to focus on referral info (which is passed in when requesting to install) or to focus on the resolution of the promise with the success/failure and `mode`?

Reply:
I've updated the text.